GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -956,10 +959,34 @@ private async Task UpdateBranchNameDisplayAsync()
             await TaskScheduler.Default;
 
             var currentBranchName = Module.GetSelectedBranch();
+            if (_branchNameLabelOnClick != null)
+            {
+                remoteNameLabel.Click -= _branchNameLabelOnClick;
+            }
+
+            var currentBranch = Module.GetRefs(false, true).FirstOrDefault(r => r.LocalName == currentBranchName);
+            if (currentBranch == null)
+            {
+                branchNameLabel.Text = currentBranchName;
+                remoteNameLabel.Text = string.Empty;
+                return;
+            }
+
+            var pushTo = string.IsNullOrEmpty(currentBranch.TrackingRemote) || string.IsNullOrEmpty(currentBranch.MergeWith)
+                ? Module.GetRemotes().Any(r => r.Name == "origin") ? $"(default) origin/{currentBranchName}" : _statusBarBranchWithoutRemote.Text
```

Comment:
`...Module.GetRemotes().Any(r => r.Name == "origin")...`
I'm afraid this is an incorrect assumption. I use "remote" and "fork" instead of "origin", and in my case I always get "_remote not configured_"

Reply:
I also think but I take the same than in the FormPush.
How the FormPush is it working for you?

Reply:
I was looking at it just now...
This is how `FormPush` does it
https://github.com/gitextensions/gitextensions/blob/14e881406bd69ed6cd279906b849ccb342443f5b/GitUI/CommandsDialogs/FormPush.cs#L222-L251

Looks like we made an assumption about "origin" in the `FormPush` as well...

. . .

![image](https://user-images.githubusercontent.com/4403806/52784953-cdfd1680-30aa-11e9-9ce3-bde609269344.png)

![image](https://user-images.githubusercontent.com/4403806/52784964-dc4b3280-30aa-11e9-8307-29e0219e0d37.png)

Reply:
Yes. most of the times, it's ok because that's the remote created when you clone.
But indeed, that's not perfect (especially in your case where "default" remote is renamed) and there is, as I know, no way in git to have a "default" remote.